site/search/Autocomplete.tsx

```diff
@@ -282,31 +290,51 @@ export function Autocomplete({
             plugins: [recentSearchesPlugin],
         })
 
-        const container = document.querySelector(AUTOCOMPLETE_CONTAINER_ID)
-        if (container) {
-            const input = container.querySelector<HTMLInputElement>("input")
-            if (input) {
-                const inputId = input.id
-                const button = container.querySelector(
-                    `label[for='${inputId}'] button`
-                )
-                // Disable the button on mount. We know there's no input because the element is created by JS
-                // and thus isn't persisted between navigations
-                button?.setAttribute("disabled", "true")
+        setSearch(search)
 
-                input.addEventListener("input", () => {
-                    const isFormValid = input.checkValidity()
-                    if (isFormValid) {
-                        button?.removeAttribute("disabled")
-                    } else {
-                        button?.setAttribute("disabled", "true")
-                    }
-                })
-            }
+        const input =
+            containerRef.current.querySelector<HTMLInputElement>("input")
+        if (input) {
+            const inputId = input.id
+            const button = containerRef.current.querySelector(
+                `label[for='${inputId}'] button`
+            )
+            // Disable the button on mount. We know there's no input because the element is created by JS
+            // and thus isn't persisted between navigations
+            button?.setAttribute("disabled", "true")
+
+            input.addEventListener("input", () => {
+                const isFormValid = input.checkValidity()
+                if (isFormValid) {
+                    button?.removeAttribute("disabled")
+                } else {
+                    button?.setAttribute("disabled", "true")
+                }
+            })
         }
 
         return () => search.destroy()
-    }, [onActivate, onClose, placeholder, detachedMediaQuery, panelClassName])
+    }, [
+        onActivate,
+        onClose,
+        placeholder,
+        detachedMediaQuery,
+        panelClassName,
+        containerRef,
+    ])
+
+    // Register a global shortcut to open the search box on typing "/"
+    useEffect(() => {
```

Comment:
Nice! 👍 

But on Firefox and Chrome, it doesn't work on the homepage. The search opens, but it isn't mounted or focused correctly 🙁

Reply:
Good call, managed to fix that :)